src/librustc_mir/const_eval.rs

```diff
@@ -234,14 +236,39 @@ impl Error for ConstEvalError {
     }
 }
 
-impl<'mir, 'tcx> interpret::Machine<'mir, 'tcx> for CompileTimeEvaluator {
+// Extra machine state for CTFE, and the Machine instance
+pub struct CompileTimeInterpreter<'a, 'mir, 'tcx: 'a+'mir> {
```

Comment:
I would prefer if this was named `ConstEvalInterpreter` or `ConstEvalMachine` to keep in the spirit of `const_eval` names *while* removing "eval" from everywhere in the `interpret` module.

Reply:
Uh, you want to rename `EvalContext`? That'll have SO many conflicts. Can we do that another time?^^

Reply:
Sure, feel free to file an issue.

Reply:
Done: https://github.com/rust-lang/rust/issues/54395